tensorflow/contrib/go/tensor.go

```diff
@@ -0,0 +1,763 @@
+package tensorflow
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"reflect"
+	"runtime"
+	"unsafe"
+
+	"github.com/golang/protobuf/proto"
+
+	pb "github.com/tensorflow/tensorflow/tensorflow/contrib/go/proto"
+)
+
+import "C"
+
+const (
+	cAckByte  = 6
+	cBellByte = 7
+	cDc1      = 17
+
+	cBytesComplex64 = 8
+	cBytesFloat32   = 4
+	cBytesFloat64   = 8
+	cBytesInt16     = 2
+	cBytesInt32     = 4
+	cBytesInt64     = 8
+	cBytesUint16    = 2
+)
+
+// DataType Type of the data contained by a Tensor
+type DataType pb.DataType
+
+// TensorInt Interface to be implemented by the tensors.
+type TensorInt interface {
```

Comment:
This is the interface, I always use this notation but here the Int makes it confusing. I'm going to rename it by TensorInterface.
Actually, this interface is not in use but I didn't remove it because I think that could be nice to have it to create Mocks for some application tests.